Issue 3937 notification email

iamonuwa opened the pull request:
<!--
  Thank you for your pull request! Please review the requirements below, read through the contributor's guide,
  and ensure your pull request has fulfilled all requirements outlined by the Gitcoin Core team.

  Contributors guide: https://docs.gitcoin.co/mk_contributors/
  
  Please note that deleting or not properly filling those sections below may result in your PR being rejected.
-->

##### Description

Send weekly unread notification emails

<!-- A description of what this PR aims to solve -->

##### Checklist

<!-- For completed items, change [ ] to [x]. -->

- [x] Read and conforms to the [contributor guidelines](https://docs.gitcoin.co/mk_contributors/)
- [x] Read and conforms to the [style guidelines](https://docs.gitcoin.co/mk_styleguide/)
- [x] Linter status: 100% pass
- [x] Added relevant tests
- [ ] Tests pass and code coverage has not decreased
- [x] Changes don't break existing behavior
- [x] Commit message follows [commit guidelines](https://docs.gitcoin.co/mk_contributors/#step-4-commit)
- [x] [Code documentation](https://docs.gitcoin.co/mk_contributors/#docstrings)
- [x] Completed: Code review by core team and any requested changes

##### Affected core subsystem(s)

<!-- Provide affected core subsystem(s) (like doc, ui, crypto, etc). -->

##### Refers/Fixes
Fixes #3937 
<!--
  Link to an issue if applicable. For example:
  If your PR fixes an issue  -> Fixes: #102, #103
  If your PR refers an issue -> Refs: #101
-->

##### Testing and Sign-off

<!-- Why should the PR reviewer trust that this change doesn't break anything? How have you tested this change? -->

###### Contributor

- [x] Read and followed the [Contributor Guidelines](https://docs.gitcoin.co/mk_contributors/)
- [x] Tested all changes **locally**
- [x] Verified existing functionality
- [x] Ran `make test` and everything passed!

###### Reviewer

- [ ] Affirm contributor guidelines have been followed and requested changes made
- [ ] CI tests and linting pass
- [ ] No conflicts (migrations, files, etc)
- [ ] Regression tested against staging or local deployment

###### Funder

- [ ] Validated requested changes were made to specification
- [ ] Bounty payout released to the contributor

thelostone-mc commented:
@iamonuwa i see tests \m/ 
btw could you throw in a screenshot of the mail too ?

iamonuwa commented:
> @iamonuwa i see tests \m/
> btw could you throw in a screenshot of the mail too ?

Will send it in tomorrow, am AFK at the moment.

PixelantDesign commented:
@thelostone-mc let me know when good to pay out!

iamonuwa commented:
![image](https://user-images.githubusercontent.com/6551094/54345147-be89d280-4642-11e9-8d32-10681aa431af.png)

iamonuwa commented:
cc: @thelostone-mc

rafalkowalski commented on app/marketing/mails.py:
```diff
@@ -686,6 +685,33 @@ def weekly_recap(to_emails=None):
+        to_emails = []
+    
+    for to_email in to_emails:
+        cur_language = translation.get_language()
```
Could you execute translation.get_language() before loop?

rafalkowalski commented on app/marketing/mails.py:
```diff
@@ -686,6 +685,33 @@ def weekly_recap(to_emails=None):
+                    subject, 
+                    text, 
+                    html, 
+                    from_name="Kevin Owocki (Gitcoin.co)", 
```
```
from_name="Kevin Owocki (Gitcoin.co)"
categories=['marketing', func_name()]
```

Above value are repeated 3 times could you create const variables at the top of file.

iamonuwa replied:
Can you elaborate on this please

thelostone-mc replied:
He means the two values are used in multiple places , so he's suggesting we move it a constant and just reuse the constant.
I'm good either way so I leave it to you @iamonuwa to make the call

iamonuwa replied:
Am fine with leaving as it is.

rafalkowalski commented on app/marketing/management/commands/send_unread_notification_emails_roundup.py:
```diff
@@ -0,0 +1,110 @@
+
+
+def is_already_sent_this_week(email):
+    from marketing.models import EmailEvent
```
Could you move imports to the top?

rafalkowalski commented on app/marketing/management/commands/send_unread_notification_emails_roundup.py:
```diff
@@ -0,0 +1,110 @@
+    from marketing.models import EmailEvent
+    from django.utils import timezone
+    then = timezone.now() - timezone.timedelta(hours=12)
+    QS = EmailEvent.objects.filter(created_on__gt=then)
```
QS variable name isn't readable. Why is this variables uppercase?

rafalkowalski commented on app/marketing/management/commands/send_unread_notification_emails_roundup.py:
```diff
@@ -0,0 +1,110 @@
+            queryset = queryset.filter(email__startswith=filter_startswith)
+        queryset = queryset.order_by('email')
+        email_list = list(set(queryset.values_list('email', flat=True)))
+        # list.sort(email_list)
```
Please remove commented code

rafalkowalski commented on app/marketing/management/commands/send_unread_notification_emails_roundup.py:
```diff
@@ -0,0 +1,110 @@
+        print("got {} emails".format(len(email_list)))
+
+        counter = 0
+        for to_email in email_list:
```
Instead of counter you could use enumerate function.

iamonuwa replied:
As well as this, why enum over counter

thelostone-mc replied:
I believe @rafalkowalski meant this ->`for index, to_email in enumerate(email_list):`

That way you wouldn't have to create a counter variable and increment it as the index is already available

iamonuwa replied:
I think this is cleaner and better

thelostone-mc commented:
@iamonuwa mind addressing the comments left by @rafalkowalski + rebase with master ?  ^_^
@PixelantDesign https://github.com/gitcoinco/web/pull/3959#issuecomment-472767747 look good ?

iamonuwa commented:
> @iamonuwa mind addressing the comments left by @rafalkowalski + rebase with master ? ^_^
> @PixelantDesign [#3959 (comment)](https://github.com/gitcoinco/web/pull/3959#issuecomment-472767747) look good ?

Yes, will make changes to reflect the review

thelostone-mc commented:
@iamonuwa we've got CI failures ! mind fixing them up ?

iamonuwa commented:
> @iamonuwa we've got CI failures ! mind fixing them up ?

Aye captain!

thelostone-mc commented:
@iamonuwa lemme know when CI is green ^_^

iamonuwa commented:
> @iamonuwa lemme know when CI is green ^_^

Working on it

iamonuwa commented:
cc: @thelostone-mc

thelostone-mc commented on app/retail/emails.py:
```diff
@@ -1050,6 +1073,11 @@ def weekly_recap(request):
+def unread_notification_email_weekly_roundup(request):
+    response_html, _ = render_unread_notification_email_weekly_roundup('mark.beacom@consensys.net')
+    return HttpResponse(response_html)
+
```
we need an extra empty line (keeping it consistent )
